src/lib.rs

```diff
@@ -275,6 +298,12 @@ mod test {
     }
 
     #[test]
+    fn parse_v4_non_zero_host_bits() {
+        let cidr = Ipv4Network::from_cidr("10.1.1.1/24");
+        assert!(cidr.is_err());
```

Comment:
I don't think this should be rejected. This shows an example http://mxtoolbox.com/subnet/?filter=10.1.1.0/24&source=subnettool